extensions/amp-onetap-google/0.1/amp-onetap-google.js

```diff
@@ -0,0 +1,322 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS-IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Embeds Google One Tap
+ *
+ * Example:
+ * <code>
+ * <amp-onetap-google
+ *   layout="nodisplay"
+ *   data-iframe_url="https://www.example.com/onetap-login"
+ * </amp-onetap-google>
+ * </code>
+ */
+
+import {CSS} from '../../../build/amp-onetap-google-0.1.css';
+import {Layout} from '../../../src/layout';
+import {
+  assertDoesNotContainDisplay,
+  setStyle,
+  setStyles,
+  toggle,
+} from '../../../src/style';
+import {isExperimentOn} from '../../../src/experiments';
+import {userAssert} from '../../../src/log';
+
+/** @const {string} */
+const TAG = 'amp-onetap-google';
+
+/** @const {string} */
+const IFRAME_ID = 'amp-onetap-google-intermediate-iframe';
+
+/** @const {Object} */
+const ACTIONS = {
+  RESIZE: 'onetap_iframe_resize',
+  SKIP: 'onetap_iframe_close',
+  DONE: 'onetap_iframe_check_status',
+  SET_UI_MODE: 'onetap_iframe_ui_mode',
+  SET_TAP_OUTSIDE_MODE: 'onetap_iframe_tap_outside_mode',
+};
+
+/** @const {Object} */
+const UI_MODES = {
+  BOTTOM_SHEET: 'bottomsheet',
+  CARD: 'card',
+};
+
+/** @const {Object} */
+const cardIframeStyle = {
+  border: 'none',
+  margin: 0,
+  padding: 0,
+  position: 'fixed',
+  right: '0',
+  left: 'auto',
+  top: '0',
+  bottom: 'auto',
+};
+
+/** @const {Object} */
+const bottomsheetIframeStyle = {
+  border: 'none',
+  margin: 0,
+  padding: 0,
+  position: 'fixed',
+  right: 'auto',
+  left: '0',
+  top: 'auto',
+  bottom: '0',
+};
+
+/** @const {Object} */
+const elementStyle = {
+  height: '0',
+  width: '0',
+  'z-index': 9999,
+  border: 'none',
+  position: 'fixed',
+  left: '0',
+  bottom: '0',
+};
+
+export class AmpOnetapGoogle extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    this.checkExperimentEnabled_();
+
+    /** @private @const {string} */
+    this.intermediateIframeUrl_ = this.element.getAttribute('data-iframe_url');
+
+    /** @private @const {string} */
+    this.intermediateIframeOrigin_ = new URL(
+      this.intermediateIframeUrl_
+    ).origin;
+
+    /** @private {!Element} */
+    this.doc_ = this.element.ownerDocument;
+
+    /** @private {boolean=} */
+    this.isBottomSheetUiMode_ = undefined;
```

Comment:
Nit: we usually use `null` for unknown values:
```suggestion
    /** @private {?boolean} */
    this.isBottomSheetUiMode_ = null;
```

Reply:
done.